internal/plugin/plugin.go

```diff
@@ -291,6 +297,48 @@ func (p IncludePlugin) Matches(pluginName, pluginVersion, latestVersion string)
 	return p.version == pluginVersion
 }
 
+// GitCommit calculates the last git commit for the plugin's directory.
+// This will return an empty string if there are uncommitted changes to the plugin's directory.
+// This is used to label the built Docker image and also avoid unnecessary Docker builds.
+func (p *Plugin) GitCommit(ctx context.Context) string {
+	p.gitCommitOnce.Do(func() {
+		if gitModified, err := calculateGitModified(ctx, p.Path); err != nil {
+			log.Printf("failed to calculate git modified status: %v", err)
```

Comment:
At some point I'd like to convert all this over to `slog` but for now this will do.